csharp/examples/IoTExample/Model/SampleDataset.cs

```diff
@@ -0,0 +1,211 @@
+                {
+                    subjectId = rand.Next(1000, 2001),
+                    activityLabel = label,
+                    timestamp = unixTime++,
+                    x_Axis = rand.NextDouble(),
+                    y_Axis = rand.NextDouble(),
+                    z_Axis = rand.NextDouble(),
+                }); 
+
+                if (success)
+                {
+                    //Console.WriteLine($"Enqueue Task 0");
+                    _size++;
+                }
+                else
+                {
+                    Console.WriteLine("Producing is blocked, percent completed is: {0}%", Math.Round((double)_size / _inputs, 4) * 100);
+                }
+            }
+
+            _rows.CompleteAdding();
+        }
+
+        public async void Consume()
+        {
+            while (!_rows.IsCompleted)
+            {
+                if (!_rows.TryTake(out SensorData item, 3000))
+                {
+                    Console.WriteLine("Consuming is blocked!");
+                }
+                else
+                {
+                    //Console.WriteLine($"Dequeue Task 1");
+                    if (item != null && item.subjectId != null)
+                    {
+                        _colSubjectId.Add((int)item.subjectId);
+                        _colActivityLabel.Add(item.activityLabel);
+                        _colTimestamp.Add((long)item.timestamp);
+                        _colXAxis.Add((double)item.x_Axis);
+                        _colYAxis.Add((double)item.y_Axis);
+                        _colZAxis.Add((double)item.z_Axis);
+                    }
+                }
+
+                if (_colSubjectId.Count == _threshold)
+                {
+                    // Build a record batch using the Fluent API
+                    var recordBatch = new RecordBatch.Builder(_memoryAllocator)
+                        .Append("SubjectId", false, col => col.Int32(array => array.AppendRange(_colSubjectId)))
+                        .Append("ActivityLabel", false, col => col.String(array => array.AppendRange(_colActivityLabel)))
+                        .Append("Timestamp", false, col => col.Int64(array => array.AppendRange(_colTimestamp)))
+                        .Append("XAxis", false, col => col.Double(array => array.AppendRange(_colXAxis)))
+                        .Append("YAxis", false, col => col.Double(array => array.AppendRange(_colYAxis)))
+                        .Append("ZAxis", false, col => col.Double(array => array.AppendRange(_colZAxis)))
+                        .Build();
+
+                    _recordBatches.Add(recordBatch);
+
+                    _colSubjectId.Clear();
+                    _colActivityLabel.Clear();
+                    _colTimestamp.Clear();
+                    _colXAxis.Clear();
+                    _colYAxis.Clear();
+                    _colZAxis.Clear();
+                }
+            }
+        }
+
+        public async Task<bool> PersistData()
```

Comment:
```suggestion
        public async Task PersistData()
```

This method never returns `false`. So you can just change it to return `Task` instead.